velox/functions/lib/aggregates/tests/AggregationTestBase.cpp

```diff
@@ -999,4 +1000,96 @@ VectorPtr AggregationTestBase::testStreaming(
   return result;
 }
 
+void AggregationTestBase::testFailingAggregations(
+    const std::vector<RowVectorPtr>& data,
+    const std::vector<std::string>& groupingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::unordered_map<std::string, std::string>& config,
+    const std::string& expectedMessage) {
+  {
+    SCOPED_TRACE("Run single");
+    auto builder = PlanBuilder().values(data);
+    builder.singleAggregation(groupingKeys, aggregates);
+    AssertQueryBuilder queryBuilder(builder.planNode());
+    queryBuilder.configs(config);
+    VELOX_ASSERT_THROW(queryBuilder.copyResults(pool()), expectedMessage);
+  }
+
+  {
+    SCOPED_TRACE("Run partial + final");
+    auto builder = PlanBuilder().values(data);
+    builder.partialAggregation(groupingKeys, aggregates).finalAggregation();
+    AssertQueryBuilder queryBuilder(builder.planNode());
+    queryBuilder.configs(config);
+    VELOX_ASSERT_THROW(queryBuilder.copyResults(pool()), expectedMessage);
+  }
+
+  {
+    SCOPED_TRACE("Run partial + final with abandon partial agg");
+    auto builder = PlanBuilder().values(data);
+    builder.partialAggregation(groupingKeys, aggregates)
+        .intermediateAggregation()
+        .finalAggregation();
+    AssertQueryBuilder queryBuilder(builder.planNode());
+    queryBuilder
```

Comment:
nit: this code repeats, right? Consider extracting a helper function to avoid copy-paste

```
enableAbandonPartialAggregation(queryBuilder);
```

Reply:
Got it, done.